packages/dashboard/src/components/delivery-alert-store.tsx

```diff
@@ -0,0 +1,539 @@
+          task_id: task_id,
+        });
+        await rmf.deliveryAlertsApi?.updateDeliveryAlertActionDeliveryAlertsDeliveryAlertIdActionPost(
+          delivery_alert_id,
+          'cancelled',
+        );
+        appController.showAlert('success', 'Successfully cancelled task');
+      } catch (e) {
+        appController.showAlert('error', `Failed to cancel task: ${(e as Error).message}`);
+      }
+    },
+    [rmf, appController],
+  );
+
+  const onOverride = React.useCallback<Required<DeliveryWarningDialogProps>['onOverride']>(
+    async (delivery_alert_id, task_id) => {
+      try {
+        if (!rmf) {
+          throw new Error('delivery alert api not available');
+        }
+        await rmf.deliveryAlertsApi?.updateDeliveryAlertActionDeliveryAlertsDeliveryAlertIdActionPost(
+          delivery_alert_id,
+          'override',
+        );
+        const taskReferenceText = task_id ? `, continuing with task ${task_id}` : '';
+        appController.showAlert(
+          'success',
+          `Overriding delivery alert ${delivery_alert_id}${taskReferenceText}`,
+        );
+      } catch (e) {
+        const taskReferenceText = task_id ? ` and continue with task ${task_id}` : '';
+        appController.showAlert(
+          'error',
+          `Failed to override delivery alert ${delivery_alert_id}${taskReferenceText}: ${
+            (e as Error).message
+          }`,
+        );
+      }
+    },
+    [rmf, appController],
+  );
+
+  const onResume = React.useCallback<Required<DeliveryWarningDialogProps>['onResume']>(
+    async (delivery_alert_id, task_id) => {
+      try {
+        if (!rmf) {
+          throw new Error('delivery alert api not available');
+        }
+        await rmf.deliveryAlertsApi?.updateDeliveryAlertActionDeliveryAlertsDeliveryAlertIdActionPost(
+          delivery_alert_id,
+          'resume',
+        );
+        const taskReferenceText = task_id ? `, continuing with task ${task_id}` : '';
+        appController.showAlert(
+          'success',
+          `Resuming after delivery alert ${delivery_alert_id}${taskReferenceText}`,
+        );
+      } catch (e) {
+        const taskReferenceText = task_id ? ` ${task_id}` : '';
+        appController.showAlert(
+          'error',
+          `Failed to resume task${taskReferenceText}: ${(e as Error).message}`,
+        );
+      }
+    },
+    [rmf, appController],
+  );
+
+  // Closing on an error requires the additional step of setting the action
+  // to cancelled, as we want to ensure this error delivery dialog does not show
+  // up anymore when the dashboard is refreshed.
+  const onErrorCloseCancel = React.useCallback<Required<DeliveryErrorDialogProps>['onClose']>(
+    async (delivery_alert_id) => {
+      try {
+        if (!rmf) {
+          throw new Error('delivery alert api not available');
+        }
+        await rmf.deliveryAlertsApi?.updateDeliveryAlertActionDeliveryAlertsDeliveryAlertIdActionPost(
```

Comment:
Maybe the '?' is not necessary

Reply:
https://github.com/open-rmf/rmf-web/pull/768/commits/83666bba825ec114cc679179300587c135db958a